test/test_inverse.py

```diff
@@ -654,3 +654,111 @@ def test_KFAC_inverse_damped_torch_matvec(
+        inv_kfac_wrong.load_state_dict(state_dict)
+
+    # create new inverse KFAC and load state dict
+    inv_kfac_new = KFACInverseLinearOperator(kfac)
+    inv_kfac_new.load_state_dict(state_dict)
+
+    # check that the two inverse KFACs are equal
+    def compare_state_dicts(state_dict: dict, state_dict_new: dict):
+        assert len(state_dict) == len(state_dict_new)
+        for value, value_new in zip(state_dict.values(), state_dict_new.values()):
+            if isinstance(value, torch.Tensor):
+                assert torch.allclose(value, value_new)
+            elif isinstance(value, dict):
+                compare_state_dicts(value, value_new)
+            elif isinstance(value, tuple):
+                assert all(
+                    torch.allclose(torch.as_tensor(v), torch.as_tensor(v2))
+                    for v, v2 in zip(value, value_new)
+                )
+            else:
+                assert value == value_new
+
+    compare_state_dicts(inv_kfac.state_dict(), inv_kfac_new.state_dict())
+
+    test_mat = torch.rand(inv_kfac.shape[1])
+    report_nonclose(inv_kfac @ test_mat, inv_kfac_new @ test_mat)
+
+
+def test_KFAC_inverse_from_state_dict():
+    """Test that KFACInverseLinearOperator can be created from state dict."""
+    torch.manual_seed(0)
+    batch_size, D_in, D_out = 4, 3, 2
+    X = torch.rand(batch_size, D_in)
+    y = torch.rand(batch_size, D_out)
+    model = torch.nn.Linear(D_in, D_out)
+
+    params = list(model.parameters())
+    # create and compute KFAC
+    kfac = KFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+        loss_average=None,
+    )
+
+    # create inverse KFAC and save state dict
+    inv_kfac = KFACInverseLinearOperator(
+        kfac, damping=1e-2, use_heuristic_damping=True, retry_double_precision=False
+    )
+    state_dict = inv_kfac.state_dict()
+
+    # create new KFAC from state dict
+    inv_kfac_new = KFACInverseLinearOperator.from_state_dict(state_dict, kfac)
+
+    # check that the two inverse KFACs are equal
+    def compare_state_dicts(state_dict: dict, state_dict_new: dict):
+        assert len(state_dict) == len(state_dict_new)
+        for value, value_new in zip(state_dict.values(), state_dict_new.values()):
+            if isinstance(value, torch.Tensor):
+                assert torch.allclose(value, value_new)
+            elif isinstance(value, dict):
+                compare_state_dicts(value, value_new)
+            elif isinstance(value, tuple):
+                assert all(
+                    torch.allclose(torch.as_tensor(v), torch.as_tensor(v2))
+                    for v, v2 in zip(value, value_new)
+                )
+            else:
+                assert value == value_new
+
+    compare_state_dicts(inv_kfac.state_dict(), inv_kfac_new.state_dict())
+
+    test_mat = torch.rand(kfac.shape[1])
+    report_nonclose(inv_kfac @ test_mat, inv_kfac_new @ test_mat)
```

Comment:
Maybe a better name ist `test_vec`?

Reply:
Maybe better to compute `inv_kfac @ test_mat` before creating the `state_dict` to avoid side effects from the tested functions on `inv_kfac`?